src/config/ConfigManager.cpp

```diff
@@ -683,6 +683,12 @@ std::string CConfigManager::getMainConfigPath() {
     if (!g_pCompositor->explicitConfigPath.empty())
         return g_pCompositor->explicitConfigPath;
 
+    if (const char *cfg = getenv("HYPRLAND_CONFIG"); cfg != NULL) {
+        return cfg;
+    }
```

Comment:
```suggestion
    std::string config_env = getenv("HYPRLAND_CONFIG");
    if (!config_env.empty())
        return config_env;
```

Reply:
this will crash idiot

Reply:
We do a little trolling

Reply:
Yeah, the string could NULL when there isn't anything there so std::string has a breakdown and crashes and burns and hoo boy a ton of smoke all over the place.